lib/traffic_drain_plug.ex

```diff
@@ -0,0 +1,99 @@
+defmodule TrafficDrain.Plug do
+  @moduledoc """
+  A `Plug` specific for usage as a kubernetes readinessProbe.
+
+
+  This `Plug` only responds to HTTP requests for `/__traffic` and will usually respond with an
+  HTTP 200 - OK statuscode. Once the BEAM process this Phoenix app is running in receives a
+  `SIGTERM` signal, requests for `/__traffic` will respond with an HTTP 503 - Service Unavailable
+  statuscode.
+
+  The default waiting time is 20_000ms and can be configured using the optional keyword `:wait`
+
+  ## Example
+
+      # In lib/your_app_web/endpoint.ex, befor any other plugs
+
+      # Make the application wait for 30s before shutting down.
+      plug TrafficDrain.Plug, wait: 30_000
+
+      plug Plug.Static,
+      ...
+
+  ## Shutdown hooks
+
+  This plug relies on Botsquad's [GracefulStop library](https://github.com/botsquad/graceful_stop)
+  which you can configure to run aditional hooks before shutdown.
+
+  Each hook is identified by a 3-element `List` of which each element serves as a positioned
+  parameter to `Task.async/3` `[module, funcion, arguments]` where `module` is the modulename in
+  which to call `function` with `args`. Function is expected to be an `atom/0` and `args` is
+  expected to be a `list()`.
+
+  Hooks can be added in the configuration for the `graceful_stop` application, using the `:hooks`
+  key. The value should be a `list/0` of hooks.
+
+      # On shutdown Hookmodule.hook("arg1", "arg2") should be called
+      config :graceful_stop, :hooks, [[Hookmodule, :hook, ["arg1", "arg2"]]
+
+
+  """
+
+  import Plug.Conn
+
+  @behaviour Plug
+
+  @impl true
+  def init(opts \\ []) do
+    # Time to wait before obeying the SIGTERM
+    wait_in_ms = Keyword.get(opts, :wait, 20_000)
+
+    # GracefulStop has its own overall timeout. If that is shorter than our wait +2.5s we'll
```

Comment:
```suggestion
    # GracefulStop has its own overall timeout. If that is shorter than our wait time + 2.5 seconds we'll
```

Waarom 2.5 seconden? Ik weet dat zelf niet, maar dat lijkt me eigenlijk interessanter om in een comment te lezen dan in woorden wat 3 regels verderop in de code ook te lezen is 😄